plugins/by-name/floaterm/default.nix

```diff
@@ -1,235 +1,257 @@
+        "new"
+        "next"
+        "prev"
+        "show"
+        "toggle"
+      ];
+  optionsRenamedToSettings = [
+    "autoclose"
+    "autohide"
+    "autoinsert"
+    "borderchars"
+    "giteditor"
+    "height"
+    "opener"
+    "position"
+    "rootmarkers"
+    "shell"
+    "title"
+    "width"
+    "wintype"
+  ];
+
+  settingsOptions = {
+    shell = mkNullOrStr ''
+      Which shell should floaterm use.
+      Default value is the same as your `shell` option.
+    '';
+
+    title = defaultNullOpts.mkStr "floaterm: $1/$2" ''
+      Title format in the floating/popup terminal window.
+      If empty, the title won't be show.
+    '';
+
+    wintype = defaultNullOpts.mkStr "float" ''
+      Set it to `"split"` or `"vsplit"` if you don't want to use floating or popup window.
+    '';
+
+    width =
+      defaultNullOpts.mkNullable (with types; either ints.unsigned (numbers.between 0.0 1.0)) 0.6
+        ''
+          Width of the floaterm window.
+          It can be either an integer (number of columns) or a float between `0.0` and `1.0`.
+          In this case, the width is relative to `columns`.
+        '';
+
+    height =
+      defaultNullOpts.mkNullable (with types; either ints.unsigned (numbers.between 0.0 1.0)) 0.6
+        ''
+          Width of the floaterm window.
+          It can be either an integer (number of lines) or a float between `0.0` and `1.0`.
+          In this case, the width is relative to `lines`.
+        '';
+
+    position = mkNullOrStr ''
+      The position of the floaterm window.
+
+      It's recommended to have a look at those options meanings, e.g. `:help :leftabove`.
+
+      - If `wintype` is `"split"` or `"vsplit"`:
+        - `"leftabove"`
+        - `"aboveleft"`
+        - `"rightbelow"`
+        - `"belowright"`
+        - `"topleft"`
+        - `"botright"` (default)
+      - If `wintype` is `"float"`:
+        - `"top"`
+        - `"bottom"`
+        - `"left"`
+        - `"right"`
+        - `"topleft"`
+        - `"topright"`
+        - `"bottomleft"`
+        - `"bottomright"`
+        - `"center"` (default)
+        - `"auto"` (at the cursor place)
+    '';
+
+    borderchars = defaultNullOpts.mkStr "─│─│┌┐┘└" ''
+      8 characters of the floating window border (top, right, bottom, left, topleft, topright,
+      botright, botleft).
+    '';
+
+    rootmarkers =
+      defaultNullOpts.mkListOf types.str
+        [
+          ".project"
+          ".git"
+          ".hg"
+          ".svn"
+          ".root"
+        ]
+        ''
+          Markers used to detect the project root directory when running.
+          ```
+          :FloatermNew --cwd=<root>
+          ```
+        '';
+
+    opener = defaultNullOpts.mkStr "split" ''
+      Command used for opening a file in the outside nvim from within `:terminal`.
+
+      Available: `"edit"`, `"split"`, `"vsplit"`, `"tabe"`, `"drop"` or user-defined commands.
+    '';
+
+    autoclose = defaultNullOpts.mkEnum [ 0 1 2 ] 1 ''
```

Comment:
Thank the nixfmt gods for allowing `[ 0 1 2 ]`

Reply:
no comment.